SensorMap.cpp

```diff
@@ -38,42 +38,197 @@ double SensorMap::getDistanceLeftDown() {
+            pwm_left_ultrasonic_right_down += pwm_left_ultrasonic_right_up;
+        }
+    }
+
+    if (current_distance_left_up < kDistanceWall) {
+        const double error_left_up = control.getErrorUltrasonic(current_distance_left_up, desireUltrasonic);
+        if (error_left_up > 0) {
+            pwm_left = motors.kLimit_inf_pwm;
+            pwm_right_ultrasonic_left_up = motors.kPAdvance * error_left_up;
+            pwm_right_ultrasonic_left_up += pwm_right_ultrasonic_right_down;
+        }
+
+        else {
+            pwm_right = motors.kLimit_inf_pwm;
+            pwm_left_ultrasonic_left_up = motors.kPAdvance * error_left_up;
+            pwm_left_ultrasonic_left_up += pwm_left_ultrasonic_right_down;
+        }
+    }
+
+    if (current_distance_left_down < kDistanceWall) {
+        const double error_left_down = control.getErrorUltrasonic(current_distance_left_down, desireUltrasonic);
+        if (error_left_down > 0) {
+            pwm_right = motors.kLimit_inf_pwm;
+            pwm_left_ultrasonic_left_down = motors.kPAdvance * error_left_down;
+            pwm_left_ultrasonic_left_down += pwm_left_ultrasonic_left_up;
+        }
+
+        else {
+            pwm_left = motors.kLimit_inf_pwm;
+            pwm_right_ultrasonic_left_down = motors.kPAdvance * error_left_down;
+            pwm_right_ultrasonic_left_down += pwm_right_ultrasonic_left_up;
+        }
+  }
+  return pwm_right_ultrasonic_left_down;
+}
+
 bool SensorMap::checkWallsRight() {
-    return (getDistanceRightUp() < 15 || getDistanceRightDown() < 15);
+    return (getDistanceRightUp() < kMaxWallDistance || getDistanceRightDown() < kMaxWallDistance);
 }
 
 bool SensorMap::checkWallsLeft() {
-  return (getDistanceLeftUp() < 15 || getDistanceLeftDown() < 15);
+  return (getDistanceLeftUp() < kMaxWallDistance || getDistanceLeftDown() < kMaxWallDistance);
 }
 
 bool SensorMap::heatVictim(double desire) {
-    double current_angle_x = logic.getAngleBNOX();
+    double current_angle_x = bno_.getAngleX();
     desire = control.getDesiredAngle(desire);
     double new_desire_left = 0;
     double new_desire_right = 0;
 
-    if (temperatureCelcius(temperature_sensor_right) > 28 and temperatureCelcius(temperature_sensor_right) < 40) {
+    if (temperatureCelcius(temperature_sensor_right) > kMinimumTemperature and temperatureCelcius(temperature_sensor_right) < kMaximumTemperature) {
         motors.stopEngines();
-        logic.turnLED();
-        delay(5000);
+        bno_.turnLED();
+        delay(kWaitFiveSeconds);
         new_desire_left = control.getNewDesireLeft(desire);
         motors.turnDegrees(new_desire_left);
-        delay(2000);
+        delay(kWaitBetweenDropKit);
         dispenser.dropOneKitRight();
-        delay(2000);
+        delay(kWaitBetweenDropKit);
         new_desire_right = control.getNewDesireRight(new_desire_left);
         motors.turnDegrees(new_desire_right);
         return 1; // Victim Right
```

Comment:
This one shouldn't drop the kits, just inform that there's a victim. Because you may not want to drop a kit, because you've already done that.